### PR TITLE
Add DeepSig version 1.2.5

### PR DESCRIPTION
**Pull request by rlibouba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/deepsig/deepsig.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    <command detect_errors="exit_code"><![CDATA[
+        deepsig
+        -f'$input_fasta'
+        -o '$output'
+        -k $organism
+        -m $output_format
+        -t $threads \${GALAXY_SLOTS:-1}
+    
+    '$output'
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:19]:
```suggestion
        '$output'
```

File: tools/deepsig/deepsig.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        <param name="organism" argument="-k" type="select" label="Organism" help="Select a organism between euk for Eukaryotes, gramp for Gram-positive bacteria or gramn for Gram-negative bacteria ">
+            <option value="euk" selected="true">euk</option>
+            <option value="gramp">gramp</option>
+            <option value="gramn">gramn</option>
+        </param>
+        <param name="output_format" argument="-m" type="select" label="output_format" help="Select a output file between gff3 (default) and json ">
+            <option value="gff3" selected="true">GFF3</option>
+            <option value="json">JSON</option>
+        </param>
+        <param name="threads" argument="-t" type="integer" label="Number of threads to use" />
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:34]: Should not be available to users

File: tools/deepsig/deepsig.xml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        deepsig_  Predictor of signal peptides in proteins based on deep learning.
+        DeepSig is a novel approach to predict signal peptides in proteins based on 
+        deep learning and sequence labelling methods. The proposed approach was 
+        evaluated and compared with other available predictors, including the top-performing SignalP.
+
+        .. _deepsig: https://github.com/BolognaBiocomp/deepsig
+    ]]></help>
+
+    <citations>
+        <citation type="bibtex"><![CDATA[
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:78]: Use your macro here

File: tools/deepsig/deepsig.xml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    ]]></command>
+
+    <inputs>
+        <param name="input_fasta" argument="-f" type="data" format="fasta" label="FASTA sequence"/>
+        <param name="output" argument="-o" type="data" format="txt" label="Output tabular file"/>
+        <param name="organism" argument="-k" type="select" label="Organism" help="Select a organism between euk for Eukaryotes, gramp for Gram-positive bacteria or gramn for Gram-negative bacteria ">
+            <option value="euk" selected="true">euk</option>
+            <option value="gramp">gramp</option>
+            <option value="gramn">gramn</option>
+        </param>
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:29]:
```suggestion
        <param name="organism" argument="-k" type="select" label="Organism" help="The kingdom the sequences belong to">
            <option value="euk" selected="true">Eukaryote (euk)</option>
            <option value="gramp">Gram-positive bacteria (gramp)</option>
            <option value="gramn">Gram-negative bacteria (gramn)</option>
        </param>
```

File: tools/deepsig/deepsig.xml [hunk 5]
```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0"?>
+<tool id="deepsig" name="deepsig" version="@TOOL_VERSION@+galaxy0" profile="21.05">
+    <description>Predictor of signal peptides in proteins</description>
+    <macros>
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:4]:
```suggestion
<tool id="deepsig" name="DeepSig" version="@TOOL_VERSION@+galaxy0" profile="21.05">
    <description>signal peptides predictor</description>
    <macros>
```

**Review by abretaud — COMMENTED** [review]
A few suggestions

File: tools/deepsig/deepsig.xml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        <requirement type="package" version="@TOOL_VERSION@">deepsig</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        deepsig
+        -f'$input_fasta'
+        -o '$output'
+        -k $organism
+        -m $output_format
+        -t $threads \${GALAXY_SLOTS:-1}
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:17]:
```suggestion
        -t \${GALAXY_SLOTS:-1}
```

File: tools/deepsig/deepsig.xml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">deepsig</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        deepsig
+        -f'$input_fasta'
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:13]:
```suggestion
        -f '$input_fasta'
```

File: tools/deepsig/deepsig.xml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        -o '$output'
+        -k $organism
+        -m $output_format
+        -t $threads \${GALAXY_SLOTS:-1}
+        '$output'
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:18]: You also wrote `-o '$output'` at line 14, which one is the proper way?

File: tools/deepsig/deepsig.xml [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    ]]></command>
+
+    <inputs>
+        <param name="input_fasta" argument="-f" type="data" format="fasta" label="FASTA sequence"/>
+        <param name="organism" argument="-k" type="select" label="Organism" help="The kingdom the sequences belong to">
+            <option value="euk" selected="true">Eukaryote (euk)</option>
+            <option value="gramp">Gram-positive bacteria (gramp)</option>
+            <option value="gramn">Gram-negative bacteria (gramn)</option>
+        </param>
+        <param name="output_format" argument="-m" type="select" label="output_format" help="Select a output file between gff3 (default) and json ">
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:28]:
```suggestion
        <param name="output_format" argument="-m" type="select" label="Output format">
```

File: tools/deepsig/deepsig.xml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        </param>
+    </inputs>
+
+    <outputs>
+        <data name="output" format="txt" label="DeepSig on ${on_string}"></data>
+    </outputs>
+
+    <tests>
+        <test expect_num_outputs="1">
+            <param name="input_fasta" value="G5ED35.fasta" ftype="fasta" />
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:40]:
```suggestion
            <param name="input_fasta" value="G5ED35.fasta" />
```

File: tools/deepsig/deepsig.xml [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+            <param name="organism" value="euk"/>
+            <param name="output_format" ftype="txt" value="gff3"/>
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:42]:
```suggestion
            <param name="output_format" value="gff3"/>
```

File: tools/deepsig/deepsig.xml [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+            <output name="output" ftype="txt">
+                <assert_contents>
+                    <has_text text="sp|G5ED35|TTR52_CAEEL" />
+                    <has_text text="evidence=ECO:0000256" />
+                </assert_contents>
+            </output>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="input_fasta" value="G5ED35.fasta" ftype="fasta" />
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:51]:
```suggestion
            <param name="input_fasta" value="G5ED35.fasta" />
```

File: tools/deepsig/deepsig.xml [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+            <param name="organism" value="euk"/>
+            <param name="output_format" ftype="txt" value="json"/>
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:53]:
```suggestion
            <param name="output_format" value="json"/>
```

File: tools/deepsig/deepsig.xml [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+            <option value="gramn">Gram-negative bacteria (gramn)</option>
+        </param>
+        <param name="output_format" argument="-m" type="select" label="output_format" help="Select a output file between gff3 (default) and json ">
+            <option value="gff3" selected="true">GFF3</option>
+            <option value="json">JSON</option>
+        </param>
+    </inputs>
+
+    <outputs>
+        <data name="output" format="txt" label="DeepSig on ${on_string}"></data>
```
Comment (abretaud) [inline tools/deepsig/deepsig.xml:35]: You should adapt the format depending on the value of output_format, like explained there: https://training.galaxyproject.org/training-material/topics/dev/tutorials/tool-integration/slides.html#27

**Review by abretaud — COMMENTED** [review]
Other comments :)

File: tools/deepsig/deepsig.xml [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+        -o '$output'
+        -k $organism
+        -m $output_format
+        -t \${GALAXY_SLOTS:-1}
+        
+    ]]></command>
+
+    <inputs>
+        <param name="input_fasta" argument="-f" type="data" format="fasta" label="FASTA sequence"/>
+        <param name="output" argument="=-o" type="data" format="txt" label="Output file"/>
```
Comment (bgruening) [inline tools/deepsig/deepsig.xml:22]: This should be removed I think. The outputs are defined in the output section.

File: tools/deepsig/deepsig.xml [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+        -o '$output'
+        -k $organism
+        -m $output_format
+        -t \${GALAXY_SLOTS:-1}
+        
+    ]]></command>
+
+    <inputs>
+        <param name="input_fasta" argument="-f" type="data" format="fasta" label="FASTA sequence"/>
+        <param name="organism" argument="-k" type="select" label="Organism" help="Select a organism between euk for Eukaryotes, gramp for Gram-positive bacteria or gramn for Gram-negative bacteria ">
```
Comment (bgruening) [inline tools/deepsig/deepsig.xml:22]: You don't need to explain the help that much if you provide proper names to the user in the select box.

File: tools/deepsig/deepsig.xml [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+            <option value="euk" selected="true">euk</option>
```
Comment (bgruening) [inline tools/deepsig/deepsig.xml:23]:
```suggestion
            <option value="euk" selected="true">Eukaryotes</option>
```

**Review by bgruening — COMMENTED** [review]
* [tools/deepsig/deepsig](https://github.com/galaxyproject/tools-iuc/pull/5279/files#diff-114400ed3f41345defdd96443816945c5933f95e98800213ac19ae99a4c37954) this should be removed from this repo I assume

File: tools/deepsig/deepsig.xml (hunk 16 above)
Comment (rlibouba) [reply tools/deepsig/deepsig.xml]: In the help section, they specified "The organism to which the sequences belong"

**Review by bgruening — APPROVED** [review]
Super cool, thanks @rlibouba

**Pull request merged**